scripts/abi_check.py

```diff
@@ -68,83 +78,142 @@ def check_abi_tools_are_installed():
             if not shutil.which(command):
                 raise Exception("{} not installed, aborting".format(command))
 
-    def get_clean_worktree_for_git_revision(self, git_rev):
-        """Make a separate worktree with git_rev checked out.
+    def _get_clean_worktree_for_git_revision(self, version):
+        """Make a separate worktree with version.revision checked out.
         Do not modify the current worktree."""
-        self.log.info(
-            "Checking out git worktree for revision {}".format(git_rev)
-        )
         git_worktree_path = tempfile.mkdtemp()
+        if version.repository:
+            self.log.debug(
```

Comment:
I think this block simplifies to
```python
self.log.debug(Checking out git worktree for revision {} from {}", version.revision, version.repository)
fetch_output = subprocess.check_output(…, encoding='utf-8')
self.log.debug(fetch_output)
worktree_rev = "FETCH_HEAD"
```
Likewise for other uses of `subprocess.Popen`. In modern Python, most uses of `subprocess` boil down to `check_call` or `check_output`.